### PR TITLE
Use Anvil KSP instead of the Square KAPT one

### PR DESCRIPTION
**Pull request by jmartinesp**
## Content

Replace [Anvil](https://github.com/square/anvil) with [Anvil KSP fork](https://github.com/ZacSweers/anvil/) by ZacSweers.

Reuse the `setupAnvil` function for this.

Also add a couple of improvements to fetching the `features` and `libraries` modules for gradle configuration.

## Motivation and context

Official Anvil has been without a stable release for > 9 months now, and has several broken features.

Besides this, we get:

- Getting back configuration cache and incremental compilation.
- Being able to update to Kotlin 2.0+.
- No more F-Droid reproducible build issues caused by the IR code generation (hopefully).
- Not needing KAPT anymore, although we could go back to using it if performance degrades with a pure KSP approach.

## Tests

Check the project builds and runs fine.

## Tested devices

- [x] Physical
- [ ] Emulator
- OS version(s): 14

## Checklist

- [ ] Changes have been tested on an Android device or Android emulator with API 23
- [ ] UI change has been tested on both light and dark themes
- [ ] Accessibility has been taken into account. See https://github.com/element-hq/element-x-android/blob/develop/CONTRIBUTING.md#accessibility
- [x] Pull request is based on the develop branch
- [x] Pull request title will be used in the release note, it clearly define what will change for the user
- [ ] Pull request includes screenshots or videos if containing UI changes
- [ ] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
- [x] You've made a self review of your PR

### CODE REVIEW

File: plugins/src/main/kotlin/extension/DependencyHandleScope.kt [hunk 1]
```diff
@@ -113,37 +92,26 @@ fun DependencyHandlerScope.allLibrariesImpl() {
...
-                implementation(project(":services:analyticsproviders:sentry"))
-            }
-        }
-    }
+    // For analytics configuration, either use noop, or use the impl, with at least one analyticsproviders implementation
+    // implementation(project(":services:analytics:noop"))
+    implementation(project(":services:analytics:impl"))
+    implementation(project(":services:analyticsproviders:posthog"))
+    implementation(project(":services:analyticsproviders:sentry"))
+
```
Comment (bmarty) [inline plugins/src/main/kotlin/extension/DependencyHandleScope.kt:100]: I think you want to check this change again, the current file is more recent.

Comment (jmartinesp) [reply plugins/src/main/kotlin/extension/DependencyHandleScope.kt:100]: Ah damn, I think the rebase broke it. Thanks, I'll take a look.

Comment (jmartinesp) [reply plugins/src/main/kotlin/extension/DependencyHandleScope.kt:110]: https://github.com/element-hq/element-x-android/pull/3564/commits/e88e70b8a5a48ef98dca23aba8c61631eef79217

File: app/src/main/kotlin/io/element/android/x/di/AppComponent.kt [hunk 2]
```diff
@@ -19,7 +18,7 @@ import io.element.android.libraries.di.SingleIn
 @SingleIn(AppScope::class)
 @MergeComponent(AppScope::class)
 interface AppComponent : NodeFactoriesBindings {
-    @Component.Factory
+    @MergeComponent.Factory
```
Comment (ganfra) [inline app/src/main/kotlin/io/element/android/x/di/AppComponent.kt:21]: 👍

**Review by ganfra — APPROVED** [review]
Thanks, LGTM if CI is happy!

**Pull request merged**